### PR TITLE
update fontawesome

### PR DESCRIPTION
**Pull request by illcat**
Fix problems that some users cannot access use.fontawesome.com.

### CODE REVIEW

**Comment by illcat** [comment]
@hamelsmu Review required.

**Review by david8zhang — APPROVED** [review]

**Review by david8zhang — APPROVED** [review]

File: _includes/custom-head.html [hunk 1]
```diff
@@ -7,18 +7,18 @@
 
 
 {%- include favicons.html -%}
-<link href="https://unpkg.com/@primer/css/dist/primer.css" rel="stylesheet" />
-<link rel="stylesheet" href="//use.fontawesome.com/releases/v5.0.7/css/all.css">
+<link href="https://cdn.jsdelivr.net/npm/@primer/css/dist/primer.css" rel="stylesheet" />
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.css">
```
Comment (hamelsmu) [inline _includes/custom-head.html:11]: Can someone comment or give me some background on why fontawesome is hosted under `fortawesome` with an `r` ?  It seems a bit odd, but then again I don't have any expertise in this area

File: _includes/custom-head.html [hunk 2]
```diff
@@ -7,18 +7,18 @@
...
 
 {%- if site.annotations -%}
-<script src="https://hypothes.is/embed.js" async></script>
+<script src="https://cdn.jsdelivr.net/npm/hypothesis/build/boot.js" async></script>
```
Comment (hamelsmu) [inline _includes/custom-head.html:14]: I would rather stick with the source listed in [the official docs](https://web.hypothes.is/for-publishers/#embedding).

**Review by hamelsmu — CHANGES_REQUESTED** [review]
Just have some questions, thanks

File: _includes/custom-head.html (hunk 1 above)
Comment (illcat) [reply _includes/custom-head.html:11]: Both sites are often inaccessible in my area.

File: _includes/custom-head.html (hunk 2 above)
Comment (illcat) [reply _includes/custom-head.html:14]: I can't open the web.hypothes.is website.

File: _includes/custom-head.html (hunk 1 above)
Comment (illcat) [reply _includes/custom-head.html:11]: There are several prices for the font fortawesome 5.

Comment (hamelsmu) [reply _includes/custom-head.html:11]: Again, what about we stick to the way listed in [the official docs](https://fontawesome.com/account/cdn):

```html
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.14.0/css/all.css" integrity="sha384-HzLeBuhoNPvSl5KYnjx0BT+WB0QEEqLprO+NBkkk5gbc67FTaL7XIGa2w1L0Xbgc" crossorigin="anonymous">
```

image

Comment (illcat) [reply _includes/custom-head.html:11]: I can't open the use.fontawesome.com，which slows down access to fastpages.

File: _includes/custom-head.html [hunk 3]
```diff
@@ -7,18 +7,18 @@
...
 {%- endif -%}
 
 {% if site.use_math %}
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css" integrity="sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq" crossorigin="anonymous">
-    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"> </script>
     <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js" integrity="sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz" crossorigin="anonymous"></script>
     <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js" integrity="sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI" crossorigin="anonymous"></script>
+    <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/mathjax@2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"> </script>
```
Comment (hamelsmu) [inline _includes/custom-head.html:21]: Why only 2.7.5, why not 3.0?

from [these docs](http://docs.mathjax.org/en/latest/web/start.html?highlight=cdn#using-mathjax-from-a-content-delivery-network-cdn)
```
<script type="text/javascript" id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>
```

Comment (illcat) [reply _includes/custom-head.html:21]: 3.x is not compatible with 2.x.

File: _includes/custom-head.html (hunk 2 above)
Comment (hamelsmu) [reply _includes/custom-head.html:14]: @illcat where are you located?  I'm weary of hosting arbitrary sites I am not familiar with as it is an attack vector, so I have an aversion towards straying from the official docs

Comment (illcat) [reply _includes/custom-head.html:14]: China

**Review by hamelsmu — APPROVED** [review]

**Comment by illcat** [comment]
I have made some corrections to the above problems.

File: _includes/custom-head.html (hunk 1 above)
Comment (hamelsmu) [reply _includes/custom-head.html]: ok thanks for your patience I think fortawesome is fine.  Thanks for educating me on this

Would love to learn why some things are not accessible for some people!  How can  I test this better in the future?

File: _includes/custom-head.html (hunk 3 above)
Comment (hamelsmu) [reply _includes/custom-head.html]: ok!  Thanks

File: _includes/custom-head.html [hunk 4]
```diff
@@ -7,18 +7,18 @@
 
 
 {%- include favicons.html -%}
-<link href="https://unpkg.com/@primer/css/dist/primer.css" rel="stylesheet" />
-<link rel="stylesheet" href="//use.fontawesome.com/releases/v5.0.7/css/all.css">
+<link href="https://cdn.jsdelivr.net/npm/@primer/css/dist/primer.css" rel="stylesheet" />
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.14.0/css/all.css" integrity="sha384-HzLeBuhoNPvSl5KYnjx0BT+WB0QEEqLprO+NBkkk5gbc67FTaL7XIGa2w1L0Xbgc" crossorigin="anonymous">
```
Comment (hamelsmu) [inline _includes/custom-head.html:11]: image

Comment (hamelsmu) [reply _includes/custom-head.html:11]: Not able to access this newer CDN

**Review by hamelsmu — CHANGES_REQUESTED** [review]

File: _includes/custom-head.html (hunk 4 above)
Comment (illcat) [reply _includes/custom-head.html]: It seems that there is no way to verify sha-256 in the third-party CDN, I changed back to the previous version

Comment (illcat) [reply _includes/custom-head.html]: We hope to find a suitable solution for Chinese users.

**Comment by hamelsmu** [comment]
Notes for myself: 

- [x] Hypothesis: https://hypothes.is/embed.js vs. https://cxn.jsdelivr.net/npm/hypothesis/build/boot.js are the same ✅  
- [x] Primer: Primer sources match exactly ✅

**Comment by hamelsmu** [comment]
@illcat it seems like the files https://hypothes.is/embed.js vs. https://cdn.jsdelivr.net/npm/hypothesis/build/boot.js are different.  seems a bit suspect to me

**Comment by hamelsmu** [comment]
I'm going to revert to https://hypothes.is/embed.js until a mirror can be found that I can verify

**Comment by hamelsmu** [comment]
reverted in https://github.com/fastai/fastpages/commit/16642b4ffa5d5bd12e2de5ffa110ac437a5f2ad5

**Comment by illcat** [comment]
My negligence, you are right for safety reasons.

**Pull request merged**